src/Aspire.Hosting.Azure.CosmosDB/AzureCosmosDBResource.cs

```diff
@@ -134,7 +134,7 @@ internal ReferenceExpression GetChildConnectionString(string childResourceName,
 
         var builder = new ReferenceExpressionBuilder();
 
-        if (UseAccessKeyAuthentication)
+        if (UseAccessKeyAuthentication && !IsEmulator)
```

Comment:
@captainsafia - FYI - I found an issue with the CosmosDB connection string logic - we should be checking IsEmulator first. I added a test for this as well.

Reply:
Ah, I see. We don't handle `WithAccessKeyAuthentication` alongisde `RunAsEmulator` well.